### PR TITLE
Rendering refactor P1: Plane unification and render relaying (pictures and video included)

### PR DESCRIPTION
**Pull request by TiviPlus**
## What does the PR do
This reworks how our rendering is handled, specifically moves away from plane masters as the end solution:
image

Instead we replace plane masters rendering directly to client with planes that render multiple planes onto them as objects in order to be able to affect multiple planes while treating them as a single object. This is done by relaying the plane using a "render relay" onto a "render plate" which acts as a plane master of plane masters of sorts, and since planes are rendered onto it as single objects any filters we apply to them will render over the planes, treating them as a single unit
image

Also cleaned up unused plane masters and render targets to reduce clutter, as well as removing a useless filter that was resulting from confusion due to said clutter.

Clientside performance testing showed no significant change, no effect on serverside performance as this is clientside.

Also added the blackness plane master so it can be relayed, side effect is that it can now be used to adjust how blackness is rendered

P2 should introduce rendering one plane to multiple render_plates, but i want to get this done before I finish that,  though testing shows its feasible

### todo
- [x] Less ghetto example than grav anomaly
- [x] finish thorough testing
- [x] Finish doccing
- [x] narrator_voice — heute um 01:17 Uhr
this PR needs seriou documentation
narrator_voice — heute um 01:17 Uhr
this is our one chance to document the render pipeline in ss13
please do the best you can

## Why It's Good For The Game

Allows more advanced effects.
As an example i made a grav anomaly effect in like 30 seconds for this video i will improve it once im awake properly:
https://streamable.com/lu98dz

Documentation images should be merged here after this pr is done
https://github.com/tgstation/documentation-assets/pull/2

## Changelog

:cl:
qol: grav anomalies now have a pretty effect
refactor: Rendering has been refactored, remember to report bugs
/:cl:

### CODE REVIEW

**Comment by AMonkeyThatCodes** [comment]
image
tg after this pr

**Comment by SplinterGP** [comment]
https://www.youtube.com/watch?v=VvFybK1Rz9Y

File: code/game/objects/effects/anomalies.dm [hunk 1]
```diff
@@ -109,6 +109,16 @@
 	)
 	AddElement(/datum/element/connect_loc, loc_connections)
 
+	var/atom/movable/warp_effect = new(src)
```
Comment (Kylerace) [inline code/game/objects/effects/anomalies.dm:112]: should this be a dedicated subtype?

**Review by Kylerace — COMMENTED** [review]
great pr, makes me think about reducing the amount of screen objects though 🤔

File: code/game/objects/effects/anomalies.dm (hunk 1 above)
Comment (TiviPlus) [reply code/game/objects/effects/anomalies.dm:112]: changed

**Comment by Kylerace** [comment]
Screenshot_1325
broken for both tivi and me, no idea why

**Comment by TiviPlus** [comment]
Fixed

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -73,6 +73,15 @@
 			header = "IT'S LOOSE",
 			notify_volume = 75
 		)
+	var/atom/movable/warp_effect = new(src)
```
Comment (Kylerace) [inline code/modules/power/singularity/singularity.dm:76]: you made a subtype for this effect specifically

File: code/modules/power/singularity/singularity.dm [hunk 3]
```diff
@@ -73,6 +73,15 @@
...
+	warp_effect.plane = GRAVITY_PULSE_PLANE //mandatory
+	warp_effect.icon = 'icons/effects/light_overlays/light_352.dmi' // your icon here!!
+	warp_effect.icon_state = "light"
+	warp_effect.pixel_x = -178
+	warp_effect.pixel_y = -178
+	animate(warp_effect, time = 2 SECONDS, loop = -1, transform = matrix().Scale(2,2))
+	animate(time = 2 SECONDS, transform = matrix())
+	vis_contents += warp_effect//mandatory
```
Comment (TiviPlus) [inline code/modules/power/singularity/singularity.dm:84]:
```suggestion
		)
```

**Review by Mothblocks — COMMENTED** [review]
Can you see if #61672 is your fault? It's a similar bug I got until I forced all planes onto the preview.

**Review by Mothblocks — CHANGES_REQUESTED** [review]
On a round without this TM the character loaded properly, please fix that bug. I can help you navigate character_preview_view or whatever it's called if you need help.

**Comment by Mothblocks** [comment]
Another bug--sec camera consoles look borked

image

**Comment by optimumtact** [comment]
can i merge it

**Comment by TiviPlus** [comment]
Works fine apart from #61756 which has no reproduction steps

**Comment by optimumtact** [comment]
https://www.youtube.com/watch?v=hnxthIijnQ4

**Pull request merged**